### PR TITLE
zipkin-server: init at 1.28.1

### PR DESCRIPTION
**Pull request by hectorj**
###### Motivation for this change

I need to install a [Zipkin](http://zipkin.io/) server.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Result:

```
result
├── bin
│   └── zipkin-server
└── share
    └── java
        └── zipkin-server-1.28.1-exec.jar
```

I used `makeWrapper` as instructed in https://nixos.org/nixpkgs/manual/#sec-language-java (though not in `installPhase`, as this was not working for some reason).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -11556,6 +11556,8 @@ with pkgs;
   zabbix20 = callPackage ../servers/monitoring/zabbix/2.0.nix { };
   zabbix22 = callPackage ../servers/monitoring/zabbix/2.2.nix { };
 
+  zipkin = callPackage ../servers/monitoring/zipkin { };
```
Comment (hectorj) [inline pkgs/top-level/all-packages.nix]: I wasn't sure what to put here: `zipkin` or `zipkin-server` ? Should it include a version number?

File: pkgs/servers/monitoring/zipkin/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    mkdir -p $out/share/java
+    cp ${src} $out/share/java/zipkin-server-${version}-exec.jar
+    mkdir -p $out/bin
+    makeWrapper ${jre}/bin/java $out/bin/zipkin \
+      --add-flags "-cp $out/share/java/zipkin-server-${version}-exec.jar org.springframework.boot.loader.JarLauncher"
+  '';
+  meta = with stdenv.lib; {
+    description = "Zipkin distributed tracing system";
+    homepage = "http://zipkin.io/";
+    license = licenses.asl20;
```
Comment (hectorj) [inline pkgs/servers/monitoring/zipkin/default.nix]: If you want to check: https://github.com/openzipkin/zipkin/blob/45600824c20f766ca6f99595df16b297ba043aa0/LICENSE

File: pkgs/servers/monitoring/zipkin/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "02369fkv0kbl1isq6y26fh2zj5wxv3zck522m5wypsjlcfcw2apa";
+  };
+  buildInputs = [ makeWrapper ];
+
+  buildCommand =
+  ''
+    mkdir -p $out/share/java
+    cp ${src} $out/share/java/zipkin-server-${version}-exec.jar
+    mkdir -p $out/bin
+    makeWrapper ${jre}/bin/java $out/bin/zipkin \
```
Comment (hectorj) [inline pkgs/servers/monitoring/zipkin/default.nix]: Maybe that should be `$out/bin/zipkin-server`? I'm really not sure

**Comment by hectorj** [comment]
I just realized I probably should have built from sources... I'm not big on Java and its build systems, so I naturally took the easy way: downloading the prebuilt JAR file.

If that is not acceptable for Nixpkgs, I can try to re-work it.

**Comment by 0xABAB** [comment]
@hectorj The consensus is that source is preferred, but I don't think there is a ban on binary packaging.

**Comment by hectorj** [comment]
I'm trying to understand why the Travis build failed, but the logs are huge and don't seem to mention `zipkin`

(but they end with `The log length has exceeded the limit of 4 MB (this usually means that the test suite is raising the same exception over and over).` so I am guessing the base build is just too verbose for Travis)

**Comment by 0xABAB** [comment]
@hectorj You pretty much have to build on your own machine if you want to be sure. Travis problems are not new, but the consensus seems to be that it has non-negative value.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: zipkin is ok.

File: pkgs/servers/monitoring/zipkin/default.nix (hunk 3 above)
Comment (Mic92) [inline pkgs/servers/monitoring/zipkin/default.nix]: Probably naming after the jar is better (`zipkin-server`). But the best thing to do, is to stick to upstream's default, if they use wrapper on other platforms.

**Comment by Mic92** [comment]
This build runs into travis limitations. You can ignore them.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (hectorj) [inline pkgs/top-level/all-packages.nix]: oops, just saw your comments after I went with `zipkin-server`. Reverting then

Comment (hectorj) [inline pkgs/top-level/all-packages.nix]: done

File: pkgs/servers/monitoring/zipkin/default.nix (hunk 3 above)
Comment (hectorj) [inline pkgs/servers/monitoring/zipkin/default.nix]: I did not find any wrapper from upstream, so I changed to `zipkin-server`

**Pull request merged**